knime-scripting-templates/R/snippet-templates.txt

```diff
@@ -663,24 +671,27 @@ outInverted <- useInverted
 
 #check the determinant of the covariance matrix to prevent error message if determinant is too small
 if(useInverted == "auto") {
-	if(det(ncCov) < .Machine$double.eps) {
-	#if too small take inverse matrix
-	useInverted <- "true"
-	} else useInverted <- "false"
+  if(det(ncCov) < .Machine$double.eps) {
+    #if too small take inverse matrix
+    useInverted <- "true"
+  } else useInverted <- "false"
 }
 if(useInverted == "true") {
-	#if too small take inverse matrix
-	ncCov <- ginv(ncCov)
-	mahaInverted <- TRUE
+  #if too small take inverse matrix
+  ncCov <- ginv(ncCov)
+  mahaInverted <- TRUE
 } 
 if(useInverted == "false") {
-	ncCov <- cov(na.omit(array(nc)))
-	mahaInverted <- FALSE
+  ncCov <- cov(na.omit(array(nc)))
+  mahaInverted <- FALSE
 }
 
 # append the distance as additional column
-kIn$mahadist <- mahalanobis(array(kIn[featSelection]), ncMean, ncCov, inverted = mahaInverted)
-kIn$mahadistMode <- outInverted
+kIn['mahadist'] <- mahalanobis(array(kIn[featSelection]), ncMean, ncCov, inverted = mahaInverted)
+kIn['p-value'] <- pchisq(kIn$mahadist, df=nDegreesFreedom, lower.tail=FALSE)
+kIn['degrees of freedom'] <- nDegreesFreedom
+kIn['mahadist-threshold'] <- qchisq(p_thresh, nDegreesFreedom, lower.tail = FALSE)
+kIn$mahadistMode <- useInverted
```

Comment:
No clue why there was the variable outInvert before. 
Conseuqently one should also remove the line 662/670 where outInvert is defined

Reply:
The old script kept the user setting of 'use inverted' in that variable to create an output column with that value. If the user had selected 'auto' the output column contained 'auto'. But I often thought it would be nice to know if 'auto' meant 'yes' or 'no' for the calculation in a particular case. I changed the script so that this column now will contain the actual decision if the inverted covariance matrix has been used. So there is no need to keep the setting anymore. I will remove it.